src/v/resource_mgmt/scheduling_groups_probe.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "cluster/partition_leaders_table.h"
+#include "config/configuration.h"
+#include "prometheus/prometheus_sanitize.h"
+#include "resource_mgmt/cpu_scheduling.h"
+#include "ssx/metrics.h"
+
+#include <seastar/core/metrics.hh>
+
+class scheduling_groups_probe {
+public:
+    void wire_up(const scheduling_groups& scheduling_groups) {
+        if (config::shard_local_cfg().disable_public_metrics()) {
+            return;
+        }
+
+        auto groups = scheduling_groups.all_scheduling_groups();
+        for (auto&& group_ref : groups) {
+            _public_metrics.add_group(
+              prometheus_sanitize::metrics_name("scheduler"),
+              {seastar::metrics::make_counter(
+                "runtime_seconds_total",
+                [group_ref] {
+                    return group_ref.get().get_stats().runtime.count();
```

Comment:
Should this have a `std::chrono::duration_cast<std::chrono::seconds>`?

Reply:
That would be a narrowing conversion though. Do we want to report fractional seconds or integer seconds? My +1 goes to fractional as we lose a lot of resolution otherwise.

Reply:
Yeah, maybe it needs to do via a `std::chrono::duration<double, std::chrono::seconds::period>`, but you get the gist.

Reply:
Currently seastar reports std::chrono::duration<float, std::chrono::seconds::period>. I'm not sure I understand what you're suggesting. You'd like it to be double instead?

Reply:
I made seastar report sched_clock::duration (i.e. nanos) and I'm converting to "fractional seconds" (i.e. std::chrono::duration<double, std::chrono::seconds::period>) in redpanda.